### PR TITLE
Google Drive Garbage Collector

### PR DESCRIPTION
**Pull request by lasryaric**
The garbage collector is re-trigger each time the folders selection changes.

### CODE REVIEW

**Review by spolu — DISMISSED** [review]
LGTM

File: connectors/src/connectors/google_drive/temporal/activities.ts [hunk 1]
```diff
@@ -490,3 +498,48 @@ export async function getFoldersToSync(connectorId: ModelId) {
...
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const authCredentials = await getAuthObject(connector.connectionId);
+  const files = await GoogleDriveFiles.findAll({
+    where: {
+      connectorId: connectorId,
+      garbageCollectedAt: { [Op.or]: [{ [Op.lt]: new Date(startTs) }, null] },
+    },
+    limit: 100,
```
Comment (spolu) [inline connectors/src/connectors/google_drive/temporal/activities.ts:518]: What if there are more than 100 here?

Comment (spolu) [reply connectors/src/connectors/google_drive/temporal/activities.ts:518]: is it possible for garbageCollectedAt to be not less than startTs or null?

Comment (spolu) [reply connectors/src/connectors/google_drive/temporal/activities.ts:518]: From the code here I don't understand how that would be possible, so these are ~all the files no? hence there will always be more than 100?

I'm probably missing something :grimacing:

Comment (spolu) [reply connectors/src/connectors/google_drive/temporal/activities.ts:518]: Ah I see the outer loop in the workflow `googleDriveGarbageCollectorWorkflow` 

Probably worth a comment for future readers

Comment (lasryaric) [reply connectors/src/connectors/google_drive/temporal/activities.ts:518]: Sure I will add it but a lot of things are processed in batches  in connectors. It's to have a retry at a reasonable cost and still be somewhat efficient.

Comment (lasryaric) [reply connectors/src/connectors/google_drive/temporal/activities.ts:518]: We use garbageCollectedAt as a pagination mechanism.
Here is what is going on:
- Get all files synced from the connectors DB, page by page (of 100).
- We check if this file should still be synchronized by calling `isObjectInFolder(objectId, selectedFolders)`
- if we keep it, just update the garbageCollectedAt to now()
- if not, we just delete it remotely and locally.

**Pull request closed without merging**